iq/iq.go

```diff
@@ -369,20 +369,18 @@ func (i *Server) submitToThirdPartyAPI(sbom string, internalID string) (string,
 		}
 		return response.StatusURL, err
 	}
+
+	// something went wrong
 	bodyBytes, err := ioutil.ReadAll(resp.Body)
 	i.logLady.WithFields(logrus.Fields{
 		"body":        string(bodyBytes),
 		"status_code": resp.StatusCode,
 		"status":      resp.Status,
 	}).Info("Request not accepted")
-	if err != nil {
```

Comment:
Why the removal of the error check?

Reply:
> Why the removal of the error check?

Could be I'm missing something, but if the path of execution gets to this point, something has failed (e.g. wrong `resp.StatusCode`). The error check was only handling an error reading the `resp.Body`, and is basically in effectual because the `bodyBytes` return value has already been ready by the `logLady.WithFields` statement.

The real problem that leads execution to this point in the code is the `resp.StatusCode`, so I provide that info in the error that gets returned instead.

Reply:
I think that makes sense now that I look at it a bit more! Sorry, the ReadAll had me thrown off. Techicnically I think you would want to do another `if err != nil` after the `ReadAll` to catch a potential error there, but that's just a mild improvement. I think I did this originally so that we could get the server response from IQ.

Reply:
This code does feel a little odd. Added logging of any error from reading the response body, but still allowing more informative error to be returned. At least this way, it doesn't look as much like a missed check of `err`. 570eafc